File: Ser2022/Store/Store/Store.Web/Modules/Store/CounterpartyRepresentatives/CounterpartyRepresentativesRow.cs

namespace Store.Store.Entities
{
    using Serenity.ComponentModel;
    using Serenity.Data;
    using Serenity.Data.Mapping;
    using System;
    using System.ComponentModel;

    [ConnectionKey("Store"), Module("Store"), TableName("[dbo].[CounterpartyRepresentatives]")]
    [DisplayName("Counterparty Representatives"), InstanceName("Counterparty Representatives")]
    [ReadPermission(PermissionKeys.Counterparties.View)]
    [ModifyPermission(PermissionKeys.Counterparties.Modify)]
    [DeletePermission(PermissionKeys.Counterparties.Delete)]
    public sealed class CounterpartyRepresentativesRow : Row<CounterpartyRepresentativesRow.RowFields>, IIdRow
    {
        [DisplayName("Representative Id"), Column("RepresentativeID"), Identity, IdProperty]
        public Int32? RepresentativeId
        {
            get => Fields.RepresentativeId[this]; 
            set =>Fields.RepresentativeId[this] = value; 
        }

        [DisplayName("Counterparty Id"), Column("CounterpartyID"), NotNull]
        public Int32? CounterpartyId
        {
            get => Fields.CounterpartyId[this]; 
            set =>Fields.CounterpartyId[this] = value; 
        }

        [DisplayName("Employee Id"), Column("EmployeeID"), NotNull]
        public Int32? EmployeeId
        {
            get => Fields.EmployeeId[this]; 
            set =>Fields.EmployeeId[this] = value; 
        }

        public CounterpartyRepresentativesRow()
        {
        }

        public CounterpartyRepresentativesRow(RowFields Fields)
            : base(Fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field RepresentativeId;
            public Int32Field CounterpartyId;
            public Int32Field EmployeeId;
        }
    }
}

File: Ser2023/Store/Store.Web/Modules/ServerTypes/Store/CounterpartyRepresentativesRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface CounterpartyRepresentativesRow {
    RepresentativeId?: number;
    CounterpartyId?: number;
    EmployeeId?: number;
}

export abstract class CounterpartyRepresentativesRow {
    static readonly idProperty = 'RepresentativeId';
    static readonly localTextPrefix = 'Store.CounterpartyRepresentatives';
    static readonly deletePermission = 'Store:Counterparties:Delete';
    static readonly insertPermission = 'Store:Counterparties:Modify';
    static readonly readPermission = 'Store:Counterparties:View';
    static readonly updatePermission = 'Store:Counterparties:Modify';

    static readonly Fields = fieldsProxy<CounterpartyRepresentativesRow>();
}
